Make HOG visualization use midpoints of orientation bins

vnavkal opened the pull request:
## Description
The HOG visualization currently draws lines perpendicular to the edges of orientation bins.  I propose that it instead draw lines perpendicular to the bisectors of orientation bins.  This would make the visualization more closely represent the distribution of gradient directions captured by the histogram.  See #2453 for more details.

Here are the changes:
* change the visualization implementation to use midpoints of orientation bins
* indicate in the `hog` docstring exactly what the visualization depicts
* add a test case checking the orientation of a line drawn by the visualization
* decrease the cell resolution of the [image in the HOG documentation](http://scikit-image.org/docs/dev/auto_examples/features_detection/plot_hog.html).  Now that the visualization does not draw horizontal and vertical lines, the example in the documentation is a bit unclear at the original resolution.  The new visualization will look like this:
![new_doc_image](https://cloud.githubusercontent.com/assets/5924638/23138473/88bac268-f75c-11e6-8d37-7f11231daa0b.png)


## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]


## References
Closes #2453

stefanv commented on skimage/feature/_hog.py:
```diff
@@ -61,7 +61,10 @@ def hog(image, orientations=9, pixels_per_cell=(8, 8), cells_per_block=(3, 3),
-        Also return an image of the HOG.
+        Also return an image of the HOG.  For each cell and orientation bin,
+        the image contains a line segment that is centered at the cell center,
+        is perpendicular to the bisector of the orientation bin, and has
```
What does "bisector of the orientation" bin mean?  The docstring has to be readable by a person new to HoG.

vnavkal replied:
@stefanv sorry for the delay.  Do you think "midpoint of the range of angles spanned by the orientation bin" is clearer?  Let me know if you have suggestions.

It seems to me that there's no way to interpret the visualization without understanding how HoG works, so consequently there's no way to explain how to interpret it without assuming some understanding of HoG.

vnavkal replied:
@stefanv just wanted to check about this again.  Do you think the alternative I suggested would clarify the docstring sufficiently?  Thanks!

stefanv replied:
Yikes, I completely missed your update here; my apologies.  Yes, that is much better!  But is it simply the midpoint of all the angles, or a weighted midpoint?

vnavkal replied:
It's simply the midpoint.  As you can see in [this](https://github.com/scikit-image/scikit-image/issues/2453#issuecomment-272940067) example, the line segments are indeed perpendicular to the midpoints of orientation bins (they all have slope 1 or -1).

I've updated the docstring; let me know if there's anything else you'd like me to change.

pep8speaks commented:
Hello @vnavkal! Thanks for updating the PR.



 - In the file [`skimage/feature/_hog.py`](https://github.com/scikit-image/scikit-image/blob/849653cd503e4c14388e20afb6f019a9ab14262e/skimage/feature/_hog.py), following are the PEP8 issues :

> [Line 193:80](https://github.com/scikit-image/scikit-image/blob/849653cd503e4c14388e20afb6f019a9ab14262e/skimage/feature/_hog.py#L193): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)

vnavkal commented:
(I just rebased with master to fix merge conflicts in `test_hog.py` introduced by the change in https://github.com/scikit-image/scikit-image/pull/2621/files#diff-f7fbb052903c8ca5aab7b798019b73b3, and I also addressed the line length issue in a new commit.)

vnavkal commented:
I just rebased again with master to fix merge conflicts introduced by #2705, which changed the name of the `visualise` parameter to `visualize`.

@stefanv do you know what is blocking this from being merged?  Does the "mrg+1" label indicate it's awaiting one more review?

jni commented:
@vnavkal yes, that is indeed what "mrg+1" means. =) Thank you!

vnavkal commented:
👍 thanks!
